src/content/app/tools/blast/state/action-listeners/blastActionListeners.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { PayloadAction, ListenerEffectAPI } from '@reduxjs/toolkit';
+
+import config from 'config';
+
+import { submitBlast } from '../blast-api/blastApiSlice';
+import {
+  restoreBlastSubmissions,
+  updateJob
+} from 'src/content/app/tools/blast/state/blast-results/blastResultsSlice';
+
+import {
+  saveBlastSubmission,
+  updateSavedBlastJob
+} from 'src/content/app/tools/blast/services/blastStorageService';
+
+import { POLLING_INTERVAL } from './blastActionListenerConstants';
+
+import type { BlastSubmission } from '../blast-results/blastResultsSlice';
```

Comment:
Move this import to the other slice imports above?

Reply:
Ah, I see I already have imports from `blastResultsSlice`; I'll move it here. Normally though, if we don't have other imports from the same module, we tend to group purely type imports in the bottom, right above the styles.